modules/account-lib/test/unit/mpc/tss.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * @prettier
+ */
+import 'should';
+const assert = require('assert');
+
+import Eddsa from '../../../src/mpc/tss';
+
+describe('TSS EDDSA key generation and signing', () => {
+  it('should generate keys and sign message', async () => {
+    const MPC = await Eddsa();
+    const A = MPC.keyShare(1, 2, 3);
+    const B = MPC.keyShare(2, 2, 3);
+    const C = MPC.keyShare(3, 2, 3);
+
+    const A_combine = MPC.keyCombine(A.uShare, [B.yShares[1], C.yShares[1]]);
+    const B_combine = MPC.keyCombine(B.uShare, [A.yShares[2], C.yShares[2]]);
+    const C_combine = MPC.keyCombine(C.uShare, [A.yShares[3], B.yShares[3]]);
+
+    const message = 'MPC on a Friday night';
+    const message_buffer = Buffer.from(message, 'utf-8');
+    // signing with 2-3 signatures
+    let A_sign_share = MPC.signShare(message_buffer, A_combine.pShare, [A_combine.jShares[2]]);
+    let B_sign_share = MPC.signShare(message_buffer, B_combine.pShare, [B_combine.jShares[1]]);
+
+    let A_sign = MPC.sign(message_buffer, A_sign_share.xShare, [B_sign_share.rShares[1]]);
+    let B_sign = MPC.sign(message_buffer, B_sign_share.xShare, [A_sign_share.rShares[2]]);
+
+    let signature = MPC.signCombine([A_sign, B_sign]);
+    let result = MPC.verify(message_buffer, signature).toString();
+    result.should.equal(message);
+
+    // signing with 3-3 signatures
+    A_sign_share = MPC.signShare(message_buffer, A_combine.pShare, [A_combine.jShares[2], A_combine.jShares[3]]);
+    B_sign_share = MPC.signShare(message_buffer, B_combine.pShare, [B_combine.jShares[1], B_combine.jShares[3]]);
+    const C_sign_share = MPC.signShare(message_buffer, C_combine.pShare, [C_combine.jShares[1], C_combine.jShares[2]]);
+
+    A_sign = MPC.sign(message_buffer, A_sign_share.xShare, [B_sign_share.rShares[1], C_sign_share.rShares[1]]);
+    B_sign = MPC.sign(message_buffer, B_sign_share.xShare, [A_sign_share.rShares[2], C_sign_share.rShares[2]]);
+    const C_sign = MPC.sign(message_buffer, C_sign_share.xShare, [A_sign_share.rShares[3], B_sign_share.rShares[3]]);
+    signature = MPC.signCombine([A_sign, B_sign, C_sign]);
+    result = MPC.verify(message_buffer, signature).toString();
+    result.should.equal(message);
+  });
```

Comment:
Add one more test where you sign with signers at indices 1 and 3 (signers A and C). This is something we need to be careful with as it will be the most commonly used signing pair, and I even made a mistake in the HSM implementation that threw an error when signing with those indices.